test/correctness/can_use_target.cpp

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
```

Comment:
Yeah, this is kinda suboptimal either way -- I wanted something that could use both get_host_target() from libHalide and this new function I'm introducing, and mixing libHalide and runtime in one target is wonky and probably not a good example to set. I'll rework this as a Generator test with ifdef'ing.